File: frontend/src/types/LithologyAlert.ts
export interface LithologyAlert {
  frameSequence: number
  bitDepth: number
  alertLevel: 'NORMAL' | 'WARNING' | 'CRITICAL'
  lithologyType: 'SANDSTONE' | 'SHALE' | 'GAS_ZONE' | 'WATER_ZONE' | 'UNKNOWN'
  transitTimeFiltered: number
  attenuationFiltered: number
  gammaRayFiltered: number
  porosity: number
  porosityDelta: number
  message: string
}

File: backend/src/main/java/com/lwd/telemetry/lithology/LithologyAlert.java
package com.lwd.telemetry.lithology;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class LithologyAlert {

    public enum AlertLevel {
        NORMAL,
        WARNING,
        CRITICAL
    }

    public enum LithologyType {
        SANDSTONE,
        SHALE,
        GAS_ZONE,
        WATER_ZONE,
        UNKNOWN
    }

    private int frameSequence;
    private float bitDepth;
    private AlertLevel alertLevel;
    private LithologyType lithologyType;

    private float transitTimeFiltered;
    private float attenuationFiltered;
    private float gammaRayFiltered;
    private float porosity;

    private float porosityDelta;
    private String message;
}
